### PR TITLE
declare kseries as an s-finite kernel

### PR DESCRIPTION
**Pull request by affeldt-aist**
##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md`

- [x] added corresponding documentation in the headers

##### Compatibility with MathComp 2.0

- [x] I added the label `TODO: HB port` to make sure someone ports this PR to
      the `hierarchy-builder` branch **or** I already opened an issue or PR (please cross reference).

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-creating-and-review-PRs) and put a milestone if possible.

### CODE REVIEW

**Comment by t6s** [comment]
Failed to compile with error messages:
```
File "./classical/cardinality.v", line 1219, characters 0-19:
Error: Cannot apply view card_set_bijP

make[2]: *** [Makefile.coq:830: classical/cardinality.vo] Error 1
make[2]: *** [classical/cardinality.vo] Deleting file 'classical/cardinality.glob'
File "./classical/functions.v", line 2403, characters 5-25:
Error: Cannot apply view subsetP

make[2]: *** [Makefile.coq:830: classical/functions.vo] Error 1
make[2]: *** [classical/functions.vo] Deleting file 'classical/functions.glob'
make[1]: *** [Makefile.coq:409: all] Error 2
```

opam packages are installed as follows:
```
coq                         8.17.0      The Coq Proof Assistant
coq-core                    8.17.0      The Coq Proof Assistant -- Core Binaries and Tools
coq-elpi                    1.17.1      Elpi extension language for Coq
coq-hierarchy-builder       1.4.0       High level commands to declare and evolve a hierarchy based on packed classes
coq-mathcomp-algebra        1.17.0      Mathematical Components Library on Algebra
coq-mathcomp-bigenough      1.0.1       A small library to do epsilon - N reasoning
coq-mathcomp-field          1.17.0      Mathematical Components Library on Fields
coq-mathcomp-fingroup       1.17.0      Mathematical Components Library on finite groups
coq-mathcomp-finmap         1.5.2       Finite sets, finite maps, finitely supported functions
coq-mathcomp-solvable       1.17.0      Mathematical Components Library on finite groups (II)
coq-mathcomp-ssreflect      1.17.0      Small Scale Reflection
coq-stdlib                  8.17.0      The Coq Proof Assistant -- Standard Library
```

**Comment by t6s** [comment]
Never mind the previous comment. The compilation have somehow succeeded after trying once again.

**Review by t6s — CHANGES_REQUESTED** [review]
The proof script looks good. The comment in the header is the only thing I want before merging.

I also want some short, additional description of kernels and s-finite kernels in the header, but it would be another PR.

File: theories/kernel.v [hunk 1]
```diff
@@ -14,19 +14,25 @@ Require Import numfun lebesgue_measure lebesgue_integral.
...
-(*   finite_measure mu == the measure mu is finite                            *)
-(*  sfinite_measure mu == the measure mu is s-finite                          *)
 (*       R.-ker X ~> Y == kernel                                              *)
-(*             kseries == countable sum of kernels                            *)
+(*                        The HB class is Kernel.                             *)
 (*     R.-sfker X ~> Y == s-finite kernel                                     *)
+(*                        The HB class is SFiniteKernel.                      *)
+(*             kseries == countable sum of kernels                            *)
+(*                        It is declared as an instance of the structure      *)
+(*                        Kernel and SFiniteKernel.                           *)
```
[the remark this answers is not in the document]
Comment (affeldt-aist) [reply theories/kernel.v]: Indeed, the doc is a bit succinct. I have added your suggestion and updated the doc.

**Review by t6s — APPROVED** [review]

**Pull request merged**